packages/core/utils/src/dal/mikro-orm/__tests__/base-entity.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { Entity, MikroORM } from "@mikro-orm/core"
+import { BaseEntity } from "../base-entity"
+
+describe("BaseEntity", () => {
+  it("should handle the id generation using the provided prefix", async () => {
+    @Entity()
+    class Entity1 extends BaseEntity {
+      constructor() {
+        super({ prefix_id: "prod" })
+      }
+    }
+
+    const orm = await MikroORM.init({
+      entities: [Entity1],
+      dbName: "test",
+      type: "postgresql",
+    })
+
+    const manager = orm.em.fork()
+    const entity1 = manager.create(Entity1, {})
+
+    expect(entity1.id).toMatch(/prod_[0-9]/)
+
+    await orm.close()
+  })
+
+  it("should handle the id generation without a provided prefix using the first three letter of the entity lower cased", async () => {
+    @Entity()
+    class Entity1 extends BaseEntity {}
+
+    const orm = await MikroORM.init({
+      entities: [Entity1],
+      dbName: "test",
+      type: "postgresql",
+    })
+
+    const manager = orm.em.fork()
+    const entity1 = manager.create(Entity1, {})
+
+    expect(entity1.id).toMatch(/ent_[0-9]/)
+
+    await orm.close()
+  })
+
+  it("should handle the id generation without a provided prefix inferring it based on the words composing the entity name excluding model and entity as part of the name", async () => {
```

Comment:
Let's add a test with a model that has custom `OnInit` and `OnCreate` hooks, and ensure the ID is still generated

Reply:
done 👍